Sources/ApodiniGRPC/ChannelHandlers/GRPCRequestDecoder.swift

```diff
@@ -0,0 +1,192 @@
+                        } else {
+                            // There's data missing
+                            precondition(!dataFrame.endStream)
+                            state = .handlingStream(initialHeaders: initialHeaders, messageCollectionCtx: messageCtx)
+                        }
+                    }
+                    if dataFrame.endStream {
+                        context.fireChannelRead(wrapInboundOut(.closeStream))
+                        state = .ready
+                    }
+                }
+            case .fileRegion(let fileRegion):
+                fatalError("Got unexpected FileRegion when expecting ByteBuffer: \(fileRegion)")
+            }
+        case .headers(let headers):
+            logger.notice("Got some headers: \(headers) (endStream: \(headers.endStream))")
+            switch state {
+            case .ready:
+                self.state = .handlingStream(initialHeaders: headers.headers, messageCollectionCtx: nil)
+                context.fireChannelRead(wrapInboundOut(.openStream(headers.headers)))
+            case .handlingStream:
+                // NOTE that this might in fact be a valid state after all, HEADERS frames can also be sent at the end of a request,
+                // although the gRPC docs don't mention this so idk maybe they don't use that.
+                // (They do use it for responses, but that doesn't apply here...)
+                fatalError("Invalid state: received HEADERS frame when handling a stream.")
+            }
+        case let .priority(priorityData): // HTTP2Frame.StreamPriorityData
+            fatalError("Got .priority: \(priorityData)")
+        case let .rstStream(errorCode): // HTTP2ErrorCode
+            print("RECEIVED RST_STREAM (w/ error code \(errorCode). CLOSING CHANNEL")
+            context.close(mode: .all, promise: nil)
+        case let .settings(settings): // HTTP2Frame.FramePayload.Settings
+            fatalError("Got .settings: \(settings)")
+        case let .pushPromise(pushPromise): // HTTP2Frame.FramePayload.PushPromise
+            fatalError("Got .pushPromise: \(pushPromise)")
+        case let .ping(pingData, ack): // HTTP2PingData, Bool
+            fatalError("Got .ping(ack=\(ack)): \(pingData)")
+        case let .goAway(lastStreamID, errorCode, opaqueData): // HTTP2StreamID, HTTP2ErrorCode, ByteBuffer?
+            fatalError("Got .goAway(lastStreamId: \(lastStreamID), errorCode: \(errorCode), opaqueData: \(opaqueData as Any))")
+        case let .windowUpdate(windowSizeIncrement): // Int
+            fatalError("Got .windowUpdate: \(windowSizeIncrement)")
+        case let .alternativeService(origin, field): // String?, ByteBuffer?
+            fatalError("Got .alternativeService(origin: \(origin as Any), field: \(field as Any)")
+        case let .origin(origins): // [String]
+            fatalError("Got .origin(\(origins))")
+        }
```

Comment:
Maybe this is similar to the fatalErrors in the protobuf decoding.. but as far as I understand, an attacker could crash the server by sending any of these unsupported HTTP2 frames 🤔

Reply:
You're right, I forgot to replace this with a proper error-handling implementation. Now fixed